.github/workflows/system-install.yml

```diff
@@ -23,8 +23,55 @@ env:
   RUSTUP_MAX_RETRIES: 10
 
 jobs:
+  build-binaries:
+    if: github.ref == 'refs/heads/main' || contains(github.event.pull_request.labels.*.name, 'test-extra')
+    strategy:
+      matrix:
+        include:
+          # We use the large GitHub actions runners
+          # For Ubuntu and Windows, this requires Organization-level configuration
+          # See: https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners#about-ubuntu-and-windows-larger-runners
+          - {
+              os: "linux-old",
+              runner: "ubuntu-latest-large",
+              container: "debian:bullseye",
+            }
+          - { os: "linux", runner: "ubuntu-latest-large", container: null }
+          - { os: "windows", runner: "windows-latest-large", container: null }
+          - { os: "macos", runner: "macos-latest", container: null }
+      fail-fast: false
+    container: ${{ matrix.container }}
+    runs-on:
+      labels: ${{ matrix.runner }}
+    name: "cargo build | ${{ matrix.os }}"
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Install Rust toolchain"
+        if: ${{ matrix.container }}
+        run: |
+          apt-get update
+          apt-get install -y curl build-essential cmake
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+          echo "PATH=$PATH:$HOME/.cargo/bin" >> "$GITHUB_ENV"
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: "Build"
+        run: cargo build
```

Comment:
Could we do a musl build? Than it's only one linux build and we can add alpine to testing

Reply:
I'll look into it! I tried this briefly and failed but I probably was doing something wrong. I presume it doesn't, but just to confirm this should not degrade test coverage right?

Reply:
No, the libc of uv shouldn't matter at all, we check what libc the host uses. If you want i can make the musl build work, i've used musl a bunch

Reply:
Thanks! https://github.com/astral-sh/uv/pull/2370